v2/internal/store/insertionindex.go

```diff
@@ -165,6 +165,20 @@ func (ii *InsertionIndex) ForEach(f func(multihash.Multihash, uint64) error) err
 	return errr
 }
 
+func (ii *InsertionIndex) ForEachCid(f func(cid.Cid, uint64) error) error {
+	var errr error
+	ii.items.AscendGreaterOrEqual(ii.items.Min(), func(i llrb.Item) bool {
+		r := i.(recordDigest).Record
+		err := f(r.Cid, r.Offset)
+		if err != nil {
+			errr = err
+			return false
+		}
+		return true
+	})
+	return errr
```

Comment:
```suggestion
	var err error
	ii.items.AscendGreaterOrEqual(ii.items.Min(), func(i llrb.Item) bool {
		r := i.(recordDigest).Record
		err = f(r.Cid, r.Offset)
		return err == nil
	})
	return err
```

Reply:
any reason to not just use one `err`?

Reply:
No other reason that I copied the pattern of `ForEach`. I'll fix that code as well.